Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs

```diff
@@ -486,80 +486,128 @@ public IEnumerator ManipulationHandlerOneHandMoveFar()
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly");
+
+                // Switch to local rotation
+                manipHandler.UseLocalSpaceForConstraint = true;
+                // rotate with x axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.XAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(testRotation, 0, 0));
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly");
+
+                // rotate with y axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.YAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(0, testRotation, 0));
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly");
+
+                // rotate with z axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.ZAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(0, 0, testRotation));
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly");
+
+                yield return hand.SetGesture(ArticulatedHandPose.GestureId.Open);
+                yield return hand.Hide();
+            }
+            finally
+            {
+                UnityEngine.Object.Destroy(parentObject);
```

Comment:
Oh this is a cool pattern, perhaps we should recommend it in  test docs. Out of curiosity, did you add this because otherwise it was hard to debug when tests were failing?

Reply:
Well I saw that this test was not cleaning up after itself, so I made the change to clean up regardless of what happens in this method at the end of this IEnumerator. This will only be true if the caller disposes of the IEnumerator, which they should as it's expected behaviour.

Reply:
But Unity unfortunately doesn't do it, so changed the approach.